src/main/java/com/microfocus/application/automation/tools/run/RunFromCodelessBuilder.java

```diff
@@ -0,0 +1,145 @@
+import jenkins.tasks.SimpleBuildStep;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.JSONValue;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.io.PrintStream;
+
+import static com.microfocus.application.automation.tools.octane.executor.UftConstants.CODELESS_FOLDER_TEMPLATE;
+
+/**
+ * @author Itay Karo on 11/11/2021
+ */
+public class RunFromCodelessBuilder extends Builder implements SimpleBuildStep {
+
+    private static final String RUN_MSG_TEMPLATE = "Running codeless for test: %s, path: %s";
+
+    private static final String RESULT_MSG_TEMPLATE = "Codeless returned code: %d";
+
+    private static final String CODELESS_BATCH_FILE = "CodelessExecuter.bat";
+
+    @DataBoundConstructor
+    public RunFromCodelessBuilder() {
+    }
+
+    @Override
+    public void perform(@NonNull Run<?, ?> build, @NonNull FilePath workspace, @NonNull Launcher launcher, @NonNull TaskListener taskListener) throws InterruptedException, IOException {
+        FilePath parentFolder = workspace.child(String.format(CODELESS_FOLDER_TEMPLATE, build.getNumber()));
+        if (!parentFolder.exists()) { // no codeless tests to run
+            return;
+        }
+
+        FilePath[] list = parentFolder.list("mbt.json");
+        if (list.length == 0) {
+            return;
+        }
+        FilePath mbtJsonFile = list[0];
+        JSONArray mbtJsonArr = (JSONArray) JSONValue.parse(mbtJsonFile.read());
+        PrintStream out = taskListener.getLogger();
+        Result currentResult = Result.NOT_BUILT;
+
+        for (Object o : mbtJsonArr) {
+            JSONObject jsonObject = (JSONObject) o;
+            String testName = jsonObject.getAsString("testName");
+            String filePath = jsonObject.getAsString("path");
+
+            ArgumentListBuilder args = new ArgumentListBuilder();
+            args.add(CODELESS_BATCH_FILE);
+            args.add(filePath);
+
+            // Run the script on node
+            // Execution result should be 0
+            try {
+                out.printf((RUN_MSG_TEMPLATE) + "%n", testName, filePath);
+                int returnCode = launcher.launch().cmds(args).stdout(out).pwd(parentFolder).join();
+                out.printf((RESULT_MSG_TEMPLATE) + "%n", returnCode);
+                currentResult = getResultFromCodeless(returnCode, currentResult);
+            } catch (IOException e) {
+                Util.displayIOException(e, taskListener);
+                build.setResult(Result.FAILURE);
+                taskListener.error("Failed running {} with exception {}", CODELESS_BATCH_FILE, e);
+            } catch (InterruptedException e) {
+                build.setResult(Result.ABORTED);
+                taskListener.error("Failed running {} - build aborted {}", CODELESS_BATCH_FILE, e);
+            }
+        }
+
+        build.setResult(currentResult);
+    }
```

Comment:
please add logs here where its exists for some validations, also if you find other places to add logs it is welcome

Reply:
done